crates/viewer/re_recording_panel/src/recording_panel_ui.rs

```diff
@@ -0,0 +1,566 @@
+use std::sync::Arc;
+
+use egui::{RichText, Widget as _};
+
+use re_data_ui::DataUi as _;
+use re_data_ui::item_ui::{entity_db_button_ui, table_id_button_ui};
+use re_log_types::TableId;
+use re_redap_browser::{Command, EXAMPLES_ORIGIN, LOCAL_ORIGIN, RedapServers};
+use re_smart_channel::SmartChannelSource;
+use re_ui::list_item::{ItemButton as _, ItemMenuButton, LabelContent};
+use re_ui::{UiExt as _, UiLayout, icons, list_item};
+use re_viewer_context::{
+    DisplayMode, Item, RecordingOrTable, SystemCommand, SystemCommandSender as _, ViewerContext,
+};
+
+use crate::data::{
+    AppIdData, DatasetData, FailedEntryData, PartitionData, RecordingPanelData, RemoteTableData,
+    ServerData, ServerEntriesData,
+};
+
+pub fn recordings_panel_ui(
+    ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    servers: &RedapServers,
+    hide_examples: bool,
+) {
+    let recording_panel_data = RecordingPanelData::new(ctx, servers, hide_examples);
+
+    ui.panel_content(|ui| {
+        ui.panel_title_bar_with_buttons(
+            "Recordings",
+            Some(
+                "These are the Recordings currently loaded in the Viewer, organized by application",
+            ),
+            |ui| {
+                add_button_ui(ctx, ui, &recording_panel_data);
+            },
+        );
+    });
+
+    egui::ScrollArea::both()
+        .id_salt("recordings_scroll_area")
+        .auto_shrink([false, true])
+        .show(ui, |ui| {
+            ui.panel_content(|ui| {
+                re_ui::list_item::list_item_scope(ui, "recording panel", |ui| {
+                    all_sections_ui(ctx, ui, servers, &recording_panel_data);
+                });
+            });
+        });
+}
+
+fn add_button_ui(
+    ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    _recording_panel_data: &RecordingPanelData<'_>,
+) {
+    use re_ui::list_item::ItemButton as _;
+    Box::new(ItemMenuButton::new(&re_ui::icons::ADD, "Add…", |ui| {
+        if re_ui::UICommand::Open
+            .menu_button_ui(ui, ctx.command_sender())
+            .clicked()
+        {
+            ui.close();
+        }
+        if re_ui::UICommand::AddRedapServer
+            .menu_button_ui(ui, ctx.command_sender())
+            .clicked()
+        {
+            ui.close();
+        }
+
+        // Show some nice debugging tools in debug builds.
+        #[cfg(debug_assertions)]
+        {
```

Comment:
fancy!!

Reply:
they were useful to me and didnt want to kill them, so I made them a home